src/Sentry.AspNetCore/DefaultUserFactory.cs

```diff
@@ -36,20 +36,21 @@ internal class DefaultUserFactory : IUserFactory
             // Identity.Name Reads the value of: ClaimsIdentity.NameClaimType which by default is ClaimTypes.Name
             // It can be changed by the application to read a different claim though:
             var name = principal.Identity?.Name;
-            if (name != null && username != name)
+            if (name is not null && username is not name)
             {
                 username = name;
             }
 
-            // Don't create a user if all we have is his IP address
-            return email == null && id == null && username == null
+            var ipAddress = context.Connection?.RemoteIpAddress?.ToString();
```

Comment:
If the server is behind a reverse proxy it would need to consider `X-Forwarded-For`.
We need to add something to the docs to mention how this works and how a user can 'override' this (basically using a `IUserFactory`

Reply:
I think the proper way to handle this is to add `UseForwardedHeaders()` middleware as per [asp.net core docs](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-5.0) which will update `HttpContext.Connection.RemoteIpAddress`. It might be worth mentioning it in the docs though.

Reply:
Or to be more precise we should document that we get the IP Address through `HttpContext.Connection.RemoteIpAddress` and if they have a reverse proxy, they need to add a middleware to deal with that by linking to the docs:

https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-5.0#forwarded-headers-middleware-order